include/zephyr/posix/signal.h

```diff
@@ -77,23 +75,31 @@ int sigismember(const sigset_t *set, int signo);
 
 typedef int	sig_atomic_t;		/* Atomic entity type (ANSI) */
 
-typedef union sigval {
+union sigval {
 	int sival_int;
 	void *sival_ptr;
-} sigval;
+};
 
-typedef struct sigevent {
+struct sigevent {
 	int sigev_notify;
 	int sigev_signo;
-	sigval sigev_value;
-	void (*sigev_notify_function)(sigval val);
-	#ifdef CONFIG_PTHREAD_IPC
+	union sigval sigev_value;
+	void (*sigev_notify_function)(union sigval val);
 	pthread_attr_t *sigev_notify_attributes;
-	#endif
-} sigevent;
+};
+
+/* Note: signal() is intentionally unimplemented in Zephyr */
+typedef void (*sighandler_t)(int signo);
+sighandler_t signal(int signum, sighandler_t handler);
 
 #ifdef __cplusplus
 }
 #endif
 
+#else /* CONFIG_POSIX_SIGNAL */
+
+#include_next <signal.h>
```

Comment:
Why is this needed? Does `-Iinclude/zephyr/posix` get added to the compiler command line?

Reply:
> Why is this needed? Does `-Iinclude/zephyr/posix` get added to the compiler command line?

This one is actually to include the next signal.h in the standard include path, which would typically be the libc